src/server/controllers/VerificationController.ts

```diff
@@ -713,6 +851,50 @@ export default class VerificationController
     res.send(getSessionJSON(session));
   };
 
+  private addInputContractEndpoint = async (req: Request, res: Response) => {
+    this.validateRequest(req);
+
+    const address = req.body.address;
+    const chainId = req.body.chainId;
+
+    // TODO: exctract bytecode
+    const bytecode = await this.verificationService.getBytecode(
+      address,
+      chainId
+    );
+
+    const { ipfs: metadataIpfsCid } = decode(bytecode);
+
+    if (!metadataIpfsCid) {
+      throw new BadRequestError(
+        "The contract doesn't have a metadata IPFS CID"
+      );
+    }
+
+    const ipfsUrl = `${IPFS_GATEWAY}${metadataIpfsCid}`;
+    const metadataFileName = "metadata.json";
+    const retrievedMetadataText = await performFetch(ipfsUrl);
```

Comment:
If we are only using `performFetch` here without hash, why don't we just use a normal `fetch()`?